src/braket/pulse/pulse_sequence.py

```diff
@@ -350,6 +350,13 @@ def __call__(self, arg: Any = None, **kwargs) -> PulseSequence:
             param_values[str(key)] = val
         return self.make_bound_pulse_sequence(param_values)
 
+    def __eq__(self, other):
+        return (
+            isinstance(other, PulseSequence)
+            and self.parameters == other.parameters
+            and self.to_ir() == other.to_ir()
```

Comment:
I think some attributes such as waveforms within a pulse sequence can be assigned a unique id if there isn't one supplied by the user. The two waveforms could be equal except for the id being different which only shows up in the ir. Should this pulse sequences be evaluted to not being equal here?

Reply:
With the `cal`s for each of the waveforms, these will be be utilized differently. Furthermore, this is in line with our current equality check between waveforms which do look at waveform ID for comparing 2

Reply:
Originally, I was looking at using program but the `Program` class in Oqpy did not have `__eq__` implemented. There may be a different function used to check for this that I missed.